src/saml.ts

```diff
@@ -1226,12 +1226,12 @@ class SAML {
         if (!restriction.Audience || !restriction.Audience[0] || !restriction.Audience[0]._) {
           return new Error("SAML assertion AudienceRestriction has no Audience value");
         }
-        if (restriction.Audience[0]._ !== expectedAudience) {
+        if (!restriction.Audience.some(audience => audience._ === expectedAudience)) {
```

Comment:
Would it be more logical to say "if every item doesn't match then throw an error" vs. what you have here which is "if you didn't find some item that matches, throw an error".

```suggestion
        if (restriction.Audience.every(audience => audience._ !== expectedAudience)) {
```